src/registry/MergedRegistry.ts

```diff
@@ -3,22 +3,30 @@ import type { Logger } from 'pino';
 import type { ChainMap, ChainMetadata, ChainName, WarpCoreConfig } from '@hyperlane-xyz/sdk';
 import { ChainAddresses } from '../types.js';
 import { objMerge } from '../utils.js';
-import { BaseRegistry } from './BaseRegistry.js';
 import { IRegistry, RegistryContent, RegistryType } from './IRegistry.js';
 
 export interface MergedRegistryOptions {
   registries: Array<IRegistry>;
   logger?: Logger;
 }
 
-export class MergedRegistry extends BaseRegistry implements IRegistry {
+/**
+ * A registry that accepts multiple sub-registries.
+ * Read methods are performed on all sub-registries and the results are merged.
+ * Write methods are performed on all sub-registries.
+ * Can be created manually or by calling `.merge()` on an existing registry.
+ */
+export class MergedRegistry implements IRegistry {
```

Comment:
Note, this class can't extend BaseRegistry anymore because BaseRegistry uses it in `merge` but it's probably fine because this class should remain simple